File: ShootR/ShootR/Server/Abilities/AbilityHandlers/ShipAbilityHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ShootR
{
    public class ShipAbilityHandler : AbilityHandler
    {
        private Ship _me;
        public ShipAbilityHandler(Ship me)
            : base(new Boost(me.MovementController, me.Controllable))
        {
            _me = me;

            _me.OnOutOfBounds += Ability(Boost.NAME).Deactivate;
        }
    }
}

File: ShootR/ShootR/Client/Abilities/AbilityHandlers/ShipAbilityHandler.ts
/// <reference path="../Boost.ts" />
/// <reference path="AbilityHandler.ts" />
/// <reference path="../../Ship/Ship.ts" />
/// <reference path="../../../Scripts/jquery.d.ts" />

class ShipAbilityHandler extends AbilityHandler {
    private boost: Boost;

    constructor (myShip: Ship) {
        var boost: Boost = new Boost(myShip.MovementController, myShip.Controllable);
        super([boost]);

        this.boost = boost;
        $(myShip).on("OnOutOfBounds", $.proxy(this.boost.Deactivate, this.boost));
    }
}